lib/railway_ipc/publisher.rb

```diff
@@ -18,13 +18,36 @@ def initialize
     end
 
     def publish(message, published_message_store=RailwayIpc::PublishedMessage)
+      message = ensure_message_uuid(message)
```

Comment:
You've been doing too much elixir :-). You don't need to return the message. It's already mutated.

Reply:
Good call! But ugh, that feels so dirty :(